GetJobber/atlantis
# chore: Add CI jobs for new docs site & uncommitted files

## Motivations

This PR adds 2 new jobs:

* Builds the new docs site to verify it doesn't fail. This will prevent us from accidentally breaking it.
* Post-build of everything, the second new job will verify the git tree doesn't contain any uncommitted changes. This will catch cases where we forget to commit generated files, such as `package-lock.json` or the new docs site prop files.

I also fixed `npm run build` in the new site package. `npm run dev` worked, but when we run `build` it produces a production bundle and checks for both typescript and vite errors. 
* Fix for TS: https://github.com/GetJobber/atlantis/pull/2094/commits/467d64b99fe3e919ee3bb4618fdaad961ebbf6fc
* Fix for Vite: https://github.com/GetJobber/atlantis/pull/2094/commits/b86b7826280960318625442fe1caf17be33c0c34


## Changes



### Added

- New CI job to verify new docs site builds properly
- New CI job to verify we're committing generated files that are expected


### Changed

- Updated eslint at the root to report unused directives

### Removed

- Unnecessary CI step for npmrc file

### Fixed

- TypeScript build errors in the new docs site (run via `npm run build`)
- Vite build issues due to missing alias for formatters package (run via `npm run build`)
- Lint warnings 


## Testing

CI passes this PR, the jobs are working as expected. You can view the latest runs here:
* [build_new_docs](https://app.circleci.com/pipelines/github/GetJobber/atlantis/14945/workflows/f8b21659-e608-4726-99dd-b50fa0151c3e/jobs/71634)
* [check_git_status](https://app.circleci.com/pipelines/github/GetJobber/atlantis/14945/workflows/f8b21659-e608-4726-99dd-b50fa0151c3e/jobs/71639)
    * See this [comment](https://github.com/GetJobber/atlantis/pull/2094/files#r1824646134) for a failed run

To test the build commands work locally:
* pull this branch, npm ci at the root
* cd packages/site
* npm run dev - should start vite's dev server and open a browser window for the new docs site
* npm run build - should successfully build the site, output should look like this screenshot:

<img width="1256" alt="Screenshot 2024-10-31 at 8 15 57 AM" src="https://github.com/user-attachments/assets/bf841524-97e2-4085-9352-4e3f085ce557">
